Menu/Matcher/Voter/AdminVoter.php

```diff
@@ -46,19 +46,31 @@ public function setRequest($request)
     public function matchItem(ItemInterface $item)
     {
         $admin = $item->getExtra('admin');
-        $match = null;
+
         if ($admin instanceof AdminInterface
             && $admin->hasRoute('list') && $admin->hasAccess('list')
-            && $this->request && $this->request->get('_sonata_admin') == $admin->getCode()
+            && $this->request
         ) {
-            $match = true;
+            $requestCode = $this->request->get('_sonata_admin');
+
+            if ($admin->getCode() === $requestCode) {
+                return true;
+            }
+
+            if ($admin->hasChildren()) {
+                foreach ($admin->getChildren() as $child) {
+                    if ($child->getBaseCodeRoute() === $requestCode) {
+                        return true;
+                    }
+                }
+            }
         }
 
         $route = $item->getExtra('route');
         if ($route && $this->request && $route == $this->request->get('_route')) {
-            $match = true;
+            return true;
         }
 
-        return $match;
+        return null;
```

Comment:
No. See https://github.com/sonata-project/SonataAdminBundle/pull/4546/files#diff-a6ff13db35adc489b376b804122502a5L49. Also, the voting strategy of menu voter (which is hardcoded in [`Knp\Menu\Matcher\Matcher::isCurrent(...)`](https://github.com/KnpLabs/KnpMenu/blob/69d6c67fab85056ddc63ae584ca8df6249f05aaa/src/Knp/Menu/Matcher/Matcher.php#L49)) is *"ignore other voters after one has voted"*. We need to abstain so other voters can decide.
And `null` is considered as `false` if there's no other voter left.